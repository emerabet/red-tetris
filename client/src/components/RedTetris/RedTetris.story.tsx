import React from 'react';
import { storiesOf } from '@storybook/react';
import RedTetris from './RedTetris';

storiesOf('Title', module)
    .add('normal red tetris', () => (
        <RedTetris />
    ))
    .add('small red tetris', () => (
        <RedTetris additionalClassName="small" />
    ));
