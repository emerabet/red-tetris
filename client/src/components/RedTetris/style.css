.redTetris {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    margin-bottom: 10px;
    background: #ed1c5b;
    -webkit-box-shadow: 0 0 30px 20px #ed1c5b;
    box-shadow: 0 0 30px 20px #ed1c5b;
  }

  .small {
      transform: scale(0.5);
  }

  .medium {
    transform: scale(0.7);
}
 
  .box {
    position: relative;
    width: 44px;
    min-width: 44px;
    height: 55px;
    margin-left: 2px;
  }
 
  .boxs {
   position: relative;
   width: 33px;
   min-width: 33px;
   height: 55px;
   margin-left: 2px;
 }
 
  .boxCube {
    background: #fceabb; /* Old browsers */
    background: -moz-linear-gradient(-45deg, #fceabb 0%, #fccd4d 50%, #f8b500 51%, #fbdf93 100%); /* FF3.6-15 */
    background: -webkit-linear-gradient(-45deg, #fceabb 0%,#fccd4d 50%,#f8b500 51%,#fbdf93 100%); /* Chrome10-25,Safari5.1-6 */
    background: linear-gradient(135deg, #fceabb 0%,#fccd4d 50%,#f8b500 51%,#fbdf93 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
    filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#fceabb', endColorstr='#fbdf93',GradientType=1 ); /* IE6-9 fallback on horizontal gradient */
    width: 10px;
    height: 10px;
    position: absolute;
  }
 
  .boxR1 {
    top: 0px;
    left: 0;
  }
 
  .boxR2 {
    background-color: violet;
   top: 11px;
   left: 0;
 }
 
 .boxR3 {
   top: 22px;
   left: 0;
  }
 
  .boxR4 {
   top: 33px;
   left: 0;
  }
 
  .boxR5 {
   top: 44px;
   left: 0;
  }
 
  .boxR6 {
   top: 0;
   left: 11px;
  }
 
  .boxR7 {
   top: 0;
   left: 22px;
  }
 
  .boxR8 {
   top: 11px;
   left: 33px;
  }
 
  .boxR9 {
   top: 22px;
   left: 22px;
  }
 
  .boxR10 {
   top: 22px;
   left: 11px;
  }
 
  .boxR11 {
   top: 33px;
   left: 33px;
  }
 
  .boxR12 {
   top: 44px;
   left: 33px;
  }
 
  .boxE1 {
   top: 0;
   left: 0;
  }
 
  .boxE2 {
   top: 11px;
   left: 0;
  }
 
  .boxE3 {
   top: 22px;
   left: 0;
  }
 
  .boxE4 {
   top: 33px;
   left: 0;
  }
 
  .boxE5 {
   top: 44px;
   left: 0;
  }
 
  .boxE6 {
   top: 0;
   left: 11px;
  }
 
  .boxE7 {
   top: 0;
   left: 22px;
  }
 
  .boxE8 {
   top: 0;
   left: 33px;
  }
 
  .boxE9 {
   top: 22px;
   left: 11px;
  }
 
  .boxE10 {
   top: 22px;
   left: 22px;
  }
 
  .boxE11 {
   top: 44px;
   left: 11px;
  }
 
  .boxE12 {
   top: 44px;
   left: 22px;
  }
 
  .boxE13 {
   top: 44px;
   left: 33px;
  }
 
  .boxD1 {
   top: 0;
   left: 0;
  }
 
  .boxD2 {
   top: 11px;
   left: 0;
  }
 
  .boxD3 {
   top: 22px;
   left: 0;
  }
 
  .boxD4 {
   top: 33px;
   left: 0;
  }
 
  .boxD5 {
   top: 44px;
   left: 0;
  }
 
  .boxD6 {
   top: 0;
   left: 11px;
  }
 
  .boxD7 {
   top: 0;
   left: 22px;
  }
 
  .boxD8 {
   top: 11px;
   left: 33px;
  }
 
  .boxD9 {
   top: 22px;
   left: 33px;
  }
 
  .boxD10 {
   top: 33px;
   left: 33px;
  }
 
  .boxD11 {
   top: 44px;
   left: 11px;
  }
 
  .boxD12 {
   top: 44px;
   left: 22px;
  }
 
  .titleSpace {
    width: 44px;
    min-width: 22px;
  }
 
  .boxT1 {
   top: 0;
   left: 0px;
  }
 
  .boxT2 {
   top: 0px;
   left:11px;
  }
 
  .boxT3 {
   top: 0;
   left: 22px;
  }
 
  .boxT4 {
   top: 11px;
   left: 11px;
  }
 
  .boxT5 {
   top: 22px;
   left: 11px;
  }
 
  .boxT6 {
   top: 33px;
   left: 11px;
  }
 
  .boxT7 {
   top: 44px;
   left: 11px;
  }
 
  .boxI1 {
   top: 0;
   left: 0;
  }
 
  .boxI2 {
   top: 0px;
   left:11px;
  }
 
  .boxI3 {
   top: 0;
   left: 22px;
  }
 
  .boxI4 {
   top: 11px;
   left: 11px;
  }
 
  .boxI5 {
   top: 22px;
   left: 11px;
  }
 
  .boxI6 {
   top: 33px;
   left: 11px;
  }
 
  .boxI7 {
   top: 44px;
   left: 11px;
  }
 
  .boxI8 {
   top: 44px;
   left: 0;
  }
 
  .boxI9 {
   top: 44px;
   left: 22px;
  }
 
 
  .boxS1 {
   top: 0;
   left: 0;
  }
 
  .boxS2 {
   top: 0;
   left:11px;
  }
 
  .boxS3 {
   top: 0;
   left: 22px;
  }
 
  .boxS4 {
   top: 0;
   left: 33px;
  }
 
  .boxS5 {
   top: 11px;
   left: 0;
  }
 
  .boxS6 {
   top: 22px;
   left: 0;
  }
 
  .boxS7 {
   top: 22px;
   left: 11px;
  }
 
  .boxS8 {
   top: 22px;
   left: 22px;
  }
 
  .boxS9 {
   top: 22px;
   left:33px;
  }
 
  .boxS10 {
   top: 33px;
   left: 33px;
  }
 
  .boxS11 {
   top: 44px;
   left: 33px;
  }
 
  .boxS12 {
   top: 44px;
   left: 22px;
  }
 
  .boxS13 {
   top: 44px;
   left: 11px;
  }
 
  .boxS14 {
   top: 44px;
   left: 0;
  }
 
  .boxS15 {
   top: 22px;
   left:33px;
  }
 