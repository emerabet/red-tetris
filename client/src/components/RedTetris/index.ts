import RedTetris from './RedTetris';

export default RedTetris;
