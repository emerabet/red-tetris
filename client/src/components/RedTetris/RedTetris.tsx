import React, { Component, useState, useEffect } from 'react';

import './style.css';

interface RedTetrisProps {
  additionalClassName?: String;
}

const RedTetris: React.SFC<RedTetrisProps> = (props) => {
  return (
    <div
      className={`redTetris ${props.additionalClassName !== undefined
        ? props.additionalClassName : ''}`}>
      <div className="box boxR">
        <div className="boxCube boxR1"></div>
        <div className="boxCube boxR2"></div>
        <div className="boxCube boxR3"></div>
        <div className="boxCube boxR4"></div>
        <div className="boxCube boxR5"></div>
        <div className="boxCube boxR6"></div>
        <div className="boxCube boxR7"></div>
        <div className="boxCube boxR8"></div>
        <div className="boxCube boxR9"></div>
        <div className="boxCube boxR10"></div>
        <div className="boxCube boxR11"></div>
        <div className="boxCube boxR12"></div>
        <div className="boxCube boxR13"></div>
        <div className="boxCube boxR14"></div>
      </div>
      <div className="box boxE">
        <div className="boxCube boxE1"></div>
        <div className="boxCube boxE2"></div>
        <div className="boxCube boxE3"></div>
        <div className="boxCube boxE4"></div>
        <div className="boxCube boxE5"></div>
        <div className="boxCube boxE6"></div>
        <div className="boxCube boxE7"></div>
        <div className="boxCube boxE8"></div>
        <div className="boxCube boxE9"></div>
        <div className="boxCube boxE10"></div>
        <div className="boxCube boxE11"></div>
        <div className="boxCube boxE12"></div>
        <div className="boxCube boxE13"></div>
      </div>
      <div className="box boxD">
        <div className="boxCube boxD1"></div>
        <div className="boxCube boxD2"></div>
        <div className="boxCube boxD3"></div>
        <div className="boxCube boxD4"></div>
        <div className="boxCube boxD5"></div>
        <div className="boxCube boxD6"></div>
        <div className="boxCube boxD7"></div>
        <div className="boxCube boxD8"></div>
        <div className="boxCube boxD9"></div>
        <div className="boxCube boxD10"></div>
        <div className="boxCube boxD11"></div>
        <div className="boxCube boxD12"></div>
      </div>
      <div className="titleSpace">
      </div>
      <div className="boxs boxT">
        <div className="boxCube boxT1"></div>
        <div className="boxCube boxT2"></div>
        <div className="boxCube boxT3"></div>
        <div className="boxCube boxT4"></div>
        <div className="boxCube boxT5"></div>
        <div className="boxCube boxT6"></div>
        <div className="boxCube boxT7"></div>
      </div>
      <div className="box boxE">
        <div className="boxCube boxE1"></div>
        <div className="boxCube boxE2"></div>
        <div className="boxCube boxE3"></div>
        <div className="boxCube boxE4"></div>
        <div className="boxCube boxE5"></div>
        <div className="boxCube boxE6"></div>
        <div className="boxCube boxE7"></div>
        <div className="boxCube boxE8"></div>
        <div className="boxCube boxE9"></div>
        <div className="boxCube boxE10"></div>
        <div className="boxCube boxE11"></div>
        <div className="boxCube boxE12"></div>
        <div className="boxCube boxE13"></div>
      </div>
      <div className="boxs boxT">
        <div className="boxCube boxT1"></div>
        <div className="boxCube boxT2"></div>
        <div className="boxCube boxT3"></div>
        <div className="boxCube boxT4"></div>
        <div className="boxCube boxT5"></div>
        <div className="boxCube boxT6"></div>
        <div className="boxCube boxT7"></div>
      </div>
      <div className="box boxR">
        <div className="boxCube boxR1"></div>
        <div className="boxCube boxR2"></div>
        <div className="boxCube boxR3"></div>
        <div className="boxCube boxR4"></div>
        <div className="boxCube boxR5"></div>
        <div className="boxCube boxR6"></div>
        <div className="boxCube boxR7"></div>
        <div className="boxCube boxR8"></div>
        <div className="boxCube boxR9"></div>
        <div className="boxCube boxR10"></div>
        <div className="boxCube boxR11"></div>
        <div className="boxCube boxR12"></div>
        <div className="boxCube boxR13"></div>
        <div className="boxCube boxR14"></div>
      </div>
      <div className="boxs boxI">
        <div className="boxCube boxI1"></div>
        <div className="boxCube boxI2"></div>
        <div className="boxCube boxI3"></div>
        <div className="boxCube boxI4"></div>
        <div className="boxCube boxI5"></div>
        <div className="boxCube boxI6"></div>
        <div className="boxCube boxI7"></div>
        <div className="boxCube boxI8"></div>
        <div className="boxCube boxI9"></div>
      </div>
      <div className="box boxS">
        <div className="boxCube boxS1"></div>
        <div className="boxCube boxS2"></div>
        <div className="boxCube boxS3"></div>
        <div className="boxCube boxS4"></div>
        <div className="boxCube boxS5"></div>
        <div className="boxCube boxS6"></div>
        <div className="boxCube boxS7"></div>
        <div className="boxCube boxS8"></div>
        <div className="boxCube boxS9"></div>
        <div className="boxCube boxS10"></div>
        <div className="boxCube boxS11"></div>
        <div className="boxCube boxS12"></div>
        <div className="boxCube boxS13"></div>
        <div className="boxCube boxS14"></div>
      </div>
    </div>
  );
};

export default RedTetris;
