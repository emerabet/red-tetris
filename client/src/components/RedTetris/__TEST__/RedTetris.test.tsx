import React from 'react';

import renderer from 'react-test-renderer';
import RedTetris from '../RedTetris';

it('renders correctly with defaults', () => {
  const redTetris = renderer.create(<RedTetris />).toJSON();
  expect(redTetris).toMatchSnapshot();
});
