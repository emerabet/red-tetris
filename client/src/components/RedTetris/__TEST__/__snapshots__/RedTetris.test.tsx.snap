// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<div
  className="redTetris "
>
  <div
    className="box boxR"
  >
    <div
      className="boxCube boxR1"
    />
    <div
      className="boxCube boxR2"
    />
    <div
      className="boxCube boxR3"
    />
    <div
      className="boxCube boxR4"
    />
    <div
      className="boxCube boxR5"
    />
    <div
      className="boxCube boxR6"
    />
    <div
      className="boxCube boxR7"
    />
    <div
      className="boxCube boxR8"
    />
    <div
      className="boxCube boxR9"
    />
    <div
      className="boxCube boxR10"
    />
    <div
      className="boxCube boxR11"
    />
    <div
      className="boxCube boxR12"
    />
    <div
      className="boxCube boxR13"
    />
    <div
      className="boxCube boxR14"
    />
  </div>
  <div
    className="box boxE"
  >
    <div
      className="boxCube boxE1"
    />
    <div
      className="boxCube boxE2"
    />
    <div
      className="boxCube boxE3"
    />
    <div
      className="boxCube boxE4"
    />
    <div
      className="boxCube boxE5"
    />
    <div
      className="boxCube boxE6"
    />
    <div
      className="boxCube boxE7"
    />
    <div
      className="boxCube boxE8"
    />
    <div
      className="boxCube boxE9"
    />
    <div
      className="boxCube boxE10"
    />
    <div
      className="boxCube boxE11"
    />
    <div
      className="boxCube boxE12"
    />
    <div
      className="boxCube boxE13"
    />
  </div>
  <div
    className="box boxD"
  >
    <div
      className="boxCube boxD1"
    />
    <div
      className="boxCube boxD2"
    />
    <div
      className="boxCube boxD3"
    />
    <div
      className="boxCube boxD4"
    />
    <div
      className="boxCube boxD5"
    />
    <div
      className="boxCube boxD6"
    />
    <div
      className="boxCube boxD7"
    />
    <div
      className="boxCube boxD8"
    />
    <div
      className="boxCube boxD9"
    />
    <div
      className="boxCube boxD10"
    />
    <div
      className="boxCube boxD11"
    />
    <div
      className="boxCube boxD12"
    />
  </div>
  <div
    className="titleSpace"
  />
  <div
    className="boxs boxT"
  >
    <div
      className="boxCube boxT1"
    />
    <div
      className="boxCube boxT2"
    />
    <div
      className="boxCube boxT3"
    />
    <div
      className="boxCube boxT4"
    />
    <div
      className="boxCube boxT5"
    />
    <div
      className="boxCube boxT6"
    />
    <div
      className="boxCube boxT7"
    />
  </div>
  <div
    className="box boxE"
  >
    <div
      className="boxCube boxE1"
    />
    <div
      className="boxCube boxE2"
    />
    <div
      className="boxCube boxE3"
    />
    <div
      className="boxCube boxE4"
    />
    <div
      className="boxCube boxE5"
    />
    <div
      className="boxCube boxE6"
    />
    <div
      className="boxCube boxE7"
    />
    <div
      className="boxCube boxE8"
    />
    <div
      className="boxCube boxE9"
    />
    <div
      className="boxCube boxE10"
    />
    <div
      className="boxCube boxE11"
    />
    <div
      className="boxCube boxE12"
    />
    <div
      className="boxCube boxE13"
    />
  </div>
  <div
    className="boxs boxT"
  >
    <div
      className="boxCube boxT1"
    />
    <div
      className="boxCube boxT2"
    />
    <div
      className="boxCube boxT3"
    />
    <div
      className="boxCube boxT4"
    />
    <div
      className="boxCube boxT5"
    />
    <div
      className="boxCube boxT6"
    />
    <div
      className="boxCube boxT7"
    />
  </div>
  <div
    className="box boxR"
  >
    <div
      className="boxCube boxR1"
    />
    <div
      className="boxCube boxR2"
    />
    <div
      className="boxCube boxR3"
    />
    <div
      className="boxCube boxR4"
    />
    <div
      className="boxCube boxR5"
    />
    <div
      className="boxCube boxR6"
    />
    <div
      className="boxCube boxR7"
    />
    <div
      className="boxCube boxR8"
    />
    <div
      className="boxCube boxR9"
    />
    <div
      className="boxCube boxR10"
    />
    <div
      className="boxCube boxR11"
    />
    <div
      className="boxCube boxR12"
    />
    <div
      className="boxCube boxR13"
    />
    <div
      className="boxCube boxR14"
    />
  </div>
  <div
    className="boxs boxI"
  >
    <div
      className="boxCube boxI1"
    />
    <div
      className="boxCube boxI2"
    />
    <div
      className="boxCube boxI3"
    />
    <div
      className="boxCube boxI4"
    />
    <div
      className="boxCube boxI5"
    />
    <div
      className="boxCube boxI6"
    />
    <div
      className="boxCube boxI7"
    />
    <div
      className="boxCube boxI8"
    />
    <div
      className="boxCube boxI9"
    />
  </div>
  <div
    className="box boxS"
  >
    <div
      className="boxCube boxS1"
    />
    <div
      className="boxCube boxS2"
    />
    <div
      className="boxCube boxS3"
    />
    <div
      className="boxCube boxS4"
    />
    <div
      className="boxCube boxS5"
    />
    <div
      className="boxCube boxS6"
    />
    <div
      className="boxCube boxS7"
    />
    <div
      className="boxCube boxS8"
    />
    <div
      className="boxCube boxS9"
    />
    <div
      className="boxCube boxS10"
    />
    <div
      className="boxCube boxS11"
    />
    <div
      className="boxCube boxS12"
    />
    <div
      className="boxCube boxS13"
    />
    <div
      className="boxCube boxS14"
    />
  </div>
</div>
`;
